### DISCUSSION TITLE
Inject a step response as input to another step

### DISCUSSION THREAD

**Discussion by fmalk** [stepci/stepci#17] [Ideas]
Hi there,

I [posted on HN](https://news.ycombinator.com/item?id=33155555) that I worked on a similar project in the past, it had the same runner idea and YAML format ideas. I'd love to contribute to this project since it runs smoothly than mine (I had a mocha injection system to run the tests).
Anyway, this project could benefit from one of my ideas, which is to use steps as a sequence of tests, and each test could use the result of a previous test/step - so an API flow could be followed. As an example, you might want to POST an object and then test the DELETE endpoint, with a user who has permission and another flow were a user doesn't have it.
If you guys are open to that suggestion, I'd work on it as a PR. What do you think?

**Comment by cweekly** [comment]
+1 This feature would make me (and others, I'm sure) much more likely to take the time to eval this tool.

**Comment by fmalk** [comment]
It seems that [Captures](https://github.com/stepci/stepci/blob/main/examples/captures.yml) does that already, but it isn't clearly documented. The example shows how to use it in a URL but is it possible to use it in a JSON body?

**Comment by mishushakov** [comment]
Hey, thanks for your question and ideas!
Indeed, this is something we already support with captures

You can capture content from response like this:

```yaml
captures:
  id:
    jsonpath: $.id
```

Then use it in consequent steps as a value, by using variables like: `{{captures.id}}`
We still lack some documentation on variables and I'll be looking forward to fix it

I'd also add, that the implementation for variable parsing is far from perfect and could be improved
For more details, see: https://github.com/stepci/runner/issues/33

Anyways, thanks for your input! We hope you would enjoy Step CI as much as we do 😄

**Comment by sebastianwis** [comment]
Hi @cweekly and @fmalk 
Thank you for your contribution here! :blush: 
It would be amazing to welcome you to our Discord if you haven't joined it yet: 
https://discord.com/invite/KqJJzJ3BTu
This is where we keep users up to date with updates, features and bug resolutions.
